src/Client.js

```diff
@@ -14,22 +14,51 @@ class Client {
     this.options = {
       cache: options.cache || false,
       cacheTime: options.cacheTime || 60,
-      cacheSize: options.cacheSize || -1
+      cacheSize: options.cacheSize || -1,
+      rateLimit: options.rateLimit || 'AUTO'
     };
     this.key = key;
+    this.requests = 0;
+    this.lastRequestAt = 9999999999999; // Set to a large number so 1st request doesn't get rate limited already
     this._validateOptions();
+    if (this.options.rateLimit !== 'NONE') {
+      this.getKeyInfo().then(info => {
+        this.requests = info.requestsInPastMin;
+        setTimeout(this._rateLimitMonitor, 1000 * info.resetsAfter);
+      });
+    }
+  }
+
+  _rateLimitMonitor () {
+    this.requests = 0;
+    // eslint-disable-next-line no-return-assign
+    setInterval(() => this.requests = 0, 1000 * 60);
+  }
+
+  _rateLimitManager () {
+    // eslint-disable-next-line no-useless-return
+    if (this.options.rateLimit === 'AUTO' && this.requests <= 60) return;
+    if (new Date().getTime() - this.lastRequestAt >= 500) return;
+    // Wait before send, because user is on HARD RateLimit mode or AUTO, but passed 60 requests/min
+    // With rate limit set to HARD, you will never be able to pass the Ratelimit set by hypixel API if this is the only script you are using the API key with.
+    // eslint-disable-next-line promise/param-names
+    return new Promise(r => setTimeout(r, 500));
   }
 
   _validateOptions (options = this.options) {
     if (typeof options !== 'object') throw new Error(Errors.OPTIONS_MUST_BE_AN_OBJECT);
     if (typeof options.cacheTime !== 'number') throw new Error(Errors.CACHE_TIME_MUST_BE_A_NUMBER);
     if (typeof options.cacheSize !== 'number') throw new Error(Errors.CACHE_LIMIT_MUST_BE_A_NUMBER);
+    if (typeof options.rateLimit !== 'string' || !['AUTO', 'HARD', 'NONE'].includes(options.rateLimit)) throw new Error(Errors.INVALID_RATE_LIMIT_OPTION);
   }
 
   async _makeRequest (url) {
     if (!url) return;
-    if (cached.has(url)) return cached.get(url);
+    if (cached.has(url) && url !== '/key') return cached.get(url);
+    if (this.options.rateLimit !== 'NONE') await this._rateLimitManager();
     const res = await fetch(BASE_URL + url + (/\?/.test(url) ? '&' : '?') + `key=${this.key}`);
+    this.requests++;
```

Comment:
```suggestion
    if (this.options.rateLimit !== 'NONE') {
        await this._rateLimitManager();
        this.requests++;
    }
    const res = await fetch(BASE_URL + url + (/\?/.test(url) ? '&' : '?') + `key=${this.key}`);
    
```

Reply:
Otherwise this.requests would be incremented infinitely because the reset interval is only created when rateLimit options are set. And there's no point in keeping tracks of the requests if the rateLimitManager is never called